web/src/main/webapp/table.ts

```diff
@@ -772,12 +772,38 @@ export class TableView extends RemoteObject
             return "?";  // TODO
     }
 
+    public drawDataRange(canvas: HTMLCanvasElement, position: number, count: number) : void {
+        var ctx = canvas.getContext('2d');
+
+        // A little hack to make canvas look prettier on retina displays.
+        // (Source: https://coderwall.com/p/vmkk6a/how-to-make-the-canvas-not-look-like-crap-on-retina)
+        ctx.scale(2, 2);
+        canvas.style.width = '100px';
+        canvas.style.height = '20px';
+        canvas.width = 2 * 100;
+        canvas.height = 2 * 20;
+
+        let x = canvas.width * position / this.rowCount;
+        let width = Math.max(1, canvas.width * count / this.rowCount);
+        ctx.fillStyle = 'rgb(200, 200, 200)';
+        ctx.fillRect(x, 0, width, canvas.height);
+
+        ctx.textBaseline = "middle";
```

Comment:
Unfortunately, the things that are drawn *on* the canvas are not HTML elements, so their styles cannot be governed by CSS... Only the properties of the canvas itself (basically, just its dimensions) can be.
Using just HTML objects for the visualization would impose a lot of restrictions, I think. Perhaps for this case it's doable, but I reckon we really need a canvas for the more interesting visualizations.